src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/IPageParameterModelConvention.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// Allows customization of the <see cref="PageParameterModel"/>.
+    /// </summary>
+    public interface IPageParameterModelConvention : IPageConvention
```

Comment:
A @pranavkm - sorry it's been a long time since we disucssed this. Can you help me understand why this isn't literally `IParameterModelConvention`?

Reply:
`PropertyModel` has references to [`ControllerModel`](https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/PropertyModel.cs#L64). We have similar issues with the other ApplicationModel pieces which have references to ControllerModel indirectly. 

We're essentially missing a concept here a la `ActionDescriptor`.